kore/test/Test/Kore/Step/Axiom/Matcher.hs

```diff
@@ -1029,6 +1043,814 @@ test_unificationWithAppMatchOnTop =
+        [ Conditional
+            { term = ()
+            , predicate = makeTruePredicate
+            , substitution =
+                Substitution.unsafeWrap
+                    ((fmap . fmap) mkInt subst)
+            }
+        ]
+    mkInt = Int.asInternal Mock.intSort
+    mkList = List.asInternal Mock.metadataTools Mock.listSort . Seq.fromList
+    cons = Mock.concatList
+    matchList = matchDefinition `on` mkList
+    matchConcrete =
+        matchList `on` fmap mkInt
+    matchVariable var val =
+            matchList
+            (either mkVar mkInt <$> var)
+            (mkInt <$> val)
+    matchCons t1 l2 =
+        matchDefinition t1
+            . mkList
+            . fmap mkInt
+            $ l2
+
+data SetElementType concrete elem set
+    = Concrete concrete
+    | ElemVar elem
+    | SetVar set
+
+concrete :: [SetElementType concrete elem set] -> [concrete]
+concrete = concat . fmap isConcrete'
+  where
+    isConcrete' =
+        \case
+            Concrete x -> [x]
+            _ -> []
+elemVars :: [SetElementType concrete elem set] -> [elem]
+elemVars = concat . fmap isElemVar
+  where
+    isElemVar =
+        \case
+            ElemVar x -> [x]
+            _ -> []
+setVars :: [SetElementType concrete elem set] -> [set]
+setVars = concat . fmap isSetVar
+  where
+    isSetVar =
+        \case
+            SetVar x -> [x]
+            _ -> []
+
+matchingSet :: [TestTree]
+matchingSet =
+    [ testCase "empty vs empty" $ do
+        let expect = top
+        actual <- matchConcrete [] []
+        assertEqualWithExplanation "" expect actual
+    , testCase "concrete vs concrete" $ do
+        let expect = top
+        actual <- matchConcrete [1, 2, 3] [1, 2, 3]
+        assertEqualWithExplanation "" expect actual
+    , testCase "empty vs concrete" $ do
+        let expect = Nothing
+        actual <- matchConcrete [] [1, 2, 3]
+        assertEqualWithExplanation "" expect actual
+    , testCase "concrete vs empty" $ do
+        let expect = Nothing
+        actual <- matchConcrete [1, 2, 3] []
+        assertEqualWithExplanation "" expect actual
+
+    , testCase "elementVar vs singleton" $ do
+        let expect = substitution [(Mock.xInt, 1)]
+        actual <- matchVariable [ElemVar Mock.xInt] [1]
+        assertEqualWithExplanation "" expect actual
+    , testCase "{elementVar, rest} vs {set}" $ do
+        let expect = substitution [(Mock.xInt, 2)]
+        actual <- matchVariable [Concrete 1, Concrete 3, ElemVar Mock.xInt] [1, 2, 3]
+        assertEqualWithExplanation "" expect actual
+    , testCase "{elementVars, rest} vs {set}" $ do
+        let expect =
+                substitution
+                    [ (Mock.xInt, 2)
+                    , (Mock.yInt, 4)
+                    ]
+        actual <-
+            matchVariable
+                [Concrete 1, Concrete 3, ElemVar Mock.xInt, ElemVar Mock.yInt]
+                [1, 2, 3, 4]
+        assertEqualWithExplanation "" expect actual
```

Comment:
This test seems potentially fragile. Are we sure that (based on the order things are stored in and such) it will always be `x |-> 2    y |-> 4`, or could it ever be `y |-> 2    x |-> 4`?

Reply:
It could definitely change in the future, so I agree these tests are brittle. I will add an issue to refactor these tests into property tests, when we have time or when this becomes a priority.